src/main/java/duke/logic/parsers/ParserUtil.java

```diff
@@ -118,7 +264,59 @@ public static int getIntegerIndexInList(int index, int listSize, String userInpu
         } catch (ArrayIndexOutOfBoundsException e) {
             throw new QueryOutOfBoundsException(String.valueOf(index));
         }
-
         throw new QueryOutOfBoundsException("INTEGER");
     }
+
+    /**
+     * Gets the integer index at index in a String list delimited by whitespace.
+     *
+     * @param userInput The users entered command which contains details of the Itinerary.
+     * @return The Itinerary object created.
+     */
+    public static Itinerary createNewItinerary(String userInput) throws DukeException {
```

Comment:
refactor

Reply:
Will do this after fixing my command bugs. Created and issue for 1.4